internal/cortex/frontend/transport/handler.go

```diff
@@ -177,11 +177,15 @@ func (f *Handler) reportSlowQuery(r *http.Request, responseHeaders http.Header,
 		thanosTraceID = traceID
 	}
 
+	remoteUser, _, _ := r.BasicAuth()
```

Comment:
Is this the only way we can get the remote user? Are there maybe some Grafana headers we can use as fallback when basic auth is not used?

Reply:
Grafana can be but one client. It's highly possible that users may use custom clients to query some data from Thanos.

Reply:
Should we indicate in the changelog that this only works for basic auth?

Reply:
Added mention of basic auth